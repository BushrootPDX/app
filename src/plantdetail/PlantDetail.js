import React from 'react';

export default function PlantDetail( props ) {
    const {
        _id,
        name, 
        scientific_name,
        image,
        spread,
        height,
        description,
        sun_requirements,
        sowing_method
    } = props.selectedPlant;

    return (
        <div id={_id}>
            <img src={image} alt={name}/>
            <h3>{name}</h3>
            <h4>{scientific_name}</h4>
            <h5>{description}</h5>
            <h5>Size:</h5><span>{spread}in. X {height}in. </span> 
            <h5>Sunlight Preference: </h5><span>{sun_requirements}</span>
            <h5>Sowing Method: </h5><span>{sowing_method}</span>
        </div>
    );
}