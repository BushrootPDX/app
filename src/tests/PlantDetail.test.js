import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PlantDetail from '../plantdetail/PlantDetail';

const className = 'smallViewer';
const plant = {
    name: 'brian', 
    scientificName: 'planty mcplanterson',
    photoUrl: 'http://picpic.com/somepic.jpg',
    spread: '14',
    height: '24',
    zone: '8',
    harvestAmount: 3,
    sunlightReq: 'partial shade',
    waterReq: 'beer only'
};

describe('PlantDetail component', () => {
    it('renders a plant detail compoenent', () => {
        const wrapper = shallow(<PlantDetail className={className} plant={plant}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});