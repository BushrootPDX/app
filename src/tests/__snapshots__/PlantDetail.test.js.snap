// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantDetail component renders a plant detail compoenent 1`] = `
<div
  className={
    Object {
      "className": "smallViewer",
      "plant": Object {
        "harvestAmount": 3,
        "height": "24",
        "name": "brian",
        "photoUrl": "http://picpic.com/somepic.jpg",
        "scientificName": "planty mcplanterson",
        "spread": "14",
        "sunlightReq": "partial shade",
        "waterReq": "beer only",
        "zone": "8",
      },
    }
  }
>
  <img />
  <h3 />
  <h5 />
  <h5>
    Size:
  </h5>
  <span>
    in. X 
    in. 
  </span>
  <h5>
    Zone:
  </h5>
  <span />
  <h5>
    Sunlight Preference: 
  </h5>
  <span />
  <h5>
    Water Requirements: 
  </h5>
  <span />
  <h5>
    Typical Yield:  
  </h5>
  <span />
</div>
`;
